### PR TITLE
feat(#35, #48): GitHub board sync + FileVault implementation

### LINKED ISSUE #48: feat: implement FileVault — concrete vault.Open() with YAML frontmatter read/write
## Context

All Go interfaces are defined (#41) but `vault.Open()` returns `fmt.Errorf("not yet implemented")`. This is the **single blocker** for every Go CLI command — status, exec, validate, batch, watch all need a concrete Vault that reads `.forgia/` files.

## Requirement

Implement `FileVault` struct that satisfies the `Vault` interface by reading/writing YAML frontmatter from `.forgia/fd/*.md` and `.forgia/sdd/*/*.md`.

### Must have
- [ ] `vault.Open(dir)` → returns `*FileVault` after verifying `.forgia/` exists
- [ ] `ListFDs()` / `GetFD()` — walk `fd/` dir, parse YAML frontmatter
- [ ] `ListSDDs()` / `GetSDD()` — walk `sdd/FD-xxx/` dirs, parse YAML frontmatter
- [ ] `CreateFD()` / `UpdateFD()` — write YAML frontmatter + markdown body
- [ ] `CreateSDD()` / `UpdateSDD()` — write YAML frontmatter + markdown body
- [ ] `FDs()` / `SDDs()` — `iter.Seq` iterators (Go 1.23+)
- [ ] `Constitution()` — read `constitution.md` raw
- [ ] `GuardrailsRaw()` — read `deny.toml` raw bytes
- [ ] `GetArchitecture()` / `ListContexts()` / `GetContext()` — read `architecture/` YAML
- [ ] `Render()` — YAML → MD conversion (can be minimal/stub initially)
- [ ] `Init()` — scaffold `.forgia/` directory (port from Bash `cmd_init`)

### Should have
- [ ] Adapter implementing `board.VaultReader` / `board.VaultWriter` using FileVault
- [ ] `config.LoadConfig()` — TOML parsing (unblocks all commands that need config)

## Dependencies

- Vault interface: `internal/vault/vault.go` (done, #41)
- FD/SDD types: `internal/vault/fd.go`, `internal/vault/sdd.go` (done, #41)
- YAML parsing: `gopkg.in/yaml.v3` (already in go.mod)
- Frontmatter parsing: `github.com/adrg/frontmatter` or manual `---` split

## Unblocks

- **#35** — board sync (SyncFromVault/SyncToVault need real vault data)
- All future Go CLI commands: `forgia status`, `forgia exec`, `forgia validate`, etc.
- `config.LoadConfig()` (can be done in same PR or follow-up)

## References

- Bash init: `bin/forgia` lines 88-253
- Vault interface: `internal/vault/vault.go`
- Go architecture plan: `docs/go-architecture.md` (Phase 2)

### PR DESCRIPTION
**Pull request by Deepzima**
## Summary

- **GitHub Projects V2 board** (`internal/board/github.go`) — full CRUD via GraphQL + `gh` CLI, LocalBoard fallback, bidirectional sync logic
- **FileVault** (`internal/vault/file_vault.go`) — concrete `vault.Open()` replacing the stub, reads/writes `.forgia/` YAML frontmatter files
- **VaultAdapter** (`internal/boardsync/adapter.go`) — bridges FileVault to board.VaultReader/VaultWriter interfaces
- **config.LoadConfig()** — TOML parsing with config.toml + config.local.toml merge
- **Security hardening** — GraphQL variable injection prevention, path traversal validation, safe type assertions

## What's in each commit

1. `feat(#35): implement GitHub Projects V2 board with ctx propagation` — GitHubBoard with card CRUD, status field mapping, ctx on all methods, slog.With() logger, LocalBoard no-op, Resolve factory
2. `feat(#35): add VaultReader/Writer DI and sync logic` — SyncFromVault (vault→board), SyncToVault (board→vault), extractIDFromTitle helper, mock tests
3. `feat(#48): implement FileVault` — vault.Open(), FD/SDD CRUD (frontmatter .md + full .yaml), Init() scaffold, architecture/contexts YAML, constitution, guardrails, iter.Seq iterators
4. `fix(#49): review fixes + should-have features + security hardening` — GraphQL variables, comma-ok assertions, VaultAdapter, config.LoadConfig(), validateID(), 24 new tests

## Test coverage

- **Board**: 12 tests (Resolve, jsonPath, extractIDFromTitle, nil-guard sync, mock contract, ghArgs injection x3)
- **Vault**: 36 tests (CRUD, iterators, frontmatter, path traversal x6, corrupted files, body preservation, SDD YAML format, round-trip fidelity, edge cases, validateID)
- **Boardsync**: 4 tests (AllItems, UpdateStatus FD/SDD, NotFound)
- **Config**: 4 tests (defaults, team config, local override, invalid TOML)
- **Total: 56+ tests across 5 packages**

## Closes

- Closes #48 (FileVault — all must-have + should-have criteria met)
- Advances #35 (board CRUD + sync logic + adapter done; CLI command, MCP tools, Beads cache, GitHub Action remain)

## Test plan

- [x] `go build ./...` passes
- [x] `go test ./...` — 56+ tests pass across all packages
- [x] Path traversal attacks rejected (6 vectors tested)
- [x] GraphQL injection prevented (dangerous chars isolated in variables)
- [x] Corrupted files in vault don't crash (graceful skip + log)
- [x] Update operations preserve markdown body
- [x] FD/SDD round-trip: all fields survive write→read
- [ ] Manual: `vault.Open()` on a real `.forgia/` directory
- [ ] Integration: end-to-end SyncFromVault with real GitHub Project

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/boardsync/adapter.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import (
+	"context"
+	"fmt"
+
+	"github.com/Deepzima/forgia/internal/board"
+	"github.com/Deepzima/forgia/internal/vault"
+)
+
+// VaultAdapter implements board.VaultReader and board.VaultWriter using a vault.Vault.
+type VaultAdapter struct {
```
Comment (ferruvich) [inline internal/boardsync/adapter.go:20]: Let's verify the VaultAdapter satisfies VaultReader and VaultWriter

```go
var _ board.VaultReader = (*VaultAdapter)(nil)
var _ board.VaultWriter = (*VaultAdapter)(nil)
```

File: internal/board/github.go [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os/exec"
+	"strings"
+)
+
+// GitHubBoard implements ProjectBoard using GitHub Projects V2 GraphQL API via gh CLI.
```
Comment (ferruvich) [inline internal/board/github.go:19]: See my comment on the adapter for interface satisfaction check
